### PR TITLE
chore: orm test

### PR DESCRIPTION
**Pull request by hwbrzzl**
## 📑 Description

1. Simplify `database/gorm/test_utils.go`, improve the testing experience of the database package, and make preparations for removing Wire.
2. Distinguish different test models: TestModelMinimum, TestModelNormal, to improve the local testing speed. The minimum model only initials one Sqlite and two Postgres; the normal model initials one Mysql, two Postgres, one Sqlite and one Sqlserver.

## Summary by CodeRabbit

- **New Features**
	- Enhanced database connection setup for testing across multiple database types (MySQL, PostgreSQL, SQL Server, SQLite) using a unified approach with `NewTestQueries`.
	- Introduced utility functions for creating and removing database migrations based on the specified database driver.

- **Bug Fixes**
	- Improved error handling in database tests, ensuring immediate failure on errors during setup.

- **Refactor**
	- Streamlined database test setup logic, replacing Docker initialization with a more modular `TestQueries` structure for better clarity and maintainability.
	- Consolidated test suite setup methods to enhance organization and readability.

- **Chores**
	- Removed redundant Docker instances and logging from test suites to simplify the testing process.

## ✅ Checks

- [x] Added test cases for my code

### CODE REVIEW

File: database/console/migrate_command_test.go [hunk 1]
```diff
@@ -25,148 +21,18 @@ func TestMigrateCommand(t *testing.T) {
 		t.Skip("Skipping tests of using docker")
 	}
 
-	var (
-		mockConfig *configmock.Config
-		query      ormcontract.Query
-	)
+	testQueries := gorm.NewTestQueries().Queries()
```
Comment (hwbrzzl) [inline database/console/migrate_command_test.go:24]: Don't distinguish drivers, get test Queries from a unique method, to improve the testing speed when using the minimum test model.

File: database/console/test_utils.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		createMysqlMigrations()
+	case contractsorm.DriverSqlserver:
+		createSqlserverMigrations()
+	case contractsorm.DriverSqlite:
+		createSqliteMigrations()
+	}
+}
+
+func createMysqlMigrations() {
+	err := file.Create("database/migrations/20230311160527_create_agents_table.up.sql",
```
Comment (kkumar-gcc) [inline database/console/test_utils.go:22]: Can't we use newly implemented migration feature here?

File: database/gorm/test_utils.go [hunk 3]
```diff
@@ -2,522 +2,593 @@ package gorm
...
+type TestQueries struct {
+	mysqlDockers     []testing.DatabaseDriver
+	postgresDockers  []testing.DatabaseDriver
+	sqliteDockers    []testing.DatabaseDriver
+	sqlserverDockers []testing.DatabaseDriver
 }
 
-func (r *MysqlDocker) New() (orm.Query, error) {
-	r.mock()
+func NewTestQueries() *TestQueries {
```
Comment (kkumar-gcc) [inline database/gorm/test_utils.go:66]: Is this internal use only?

File: database/gorm/to_sql_test.go [hunk 4]
```diff
@@ -21,34 +20,33 @@ func TestToSqlTestSuite(t *testing.T) {
...
-		log.Fatalf("Init mysql error: %s", err)
-	}
+	suite.Run(t, &ToSqlTestSuite{})
+}
 
-	suite.Run(t, &ToSqlTestSuite{
-		query: query,
-	})
+func (s *ToSqlTestSuite) SetupSuite() {
+	postgresQuery := NewTestQuery(docker.Postgres())
```
Comment (kkumar-gcc) [inline database/gorm/to_sql_test.go:27]: Why are we only testing postgres here, shouldn't we test other driver also?

File: database/console/test_utils.go (hunk 2 above)
Comment (hwbrzzl) [reply database/console/test_utils.go:22]: Can't for now, will optimize it in another PR.

File: database/gorm/test_utils.go (hunk 3 above)
Comment (hwbrzzl) [reply database/gorm/test_utils.go:66]: No, need to be used in other framework packages, but a normal user should not use it.

File: database/gorm/to_sql_test.go (hunk 4 above)
Comment (hwbrzzl) [reply database/gorm/to_sql_test.go:27]: The main purpose is to test if the ToSql function is correct, so one driver is enough.

Comment (kkumar-gcc) [reply database/gorm/to_sql_test.go:27]: Okay

**Review by kkumar-gcc — APPROVED** [review]
Great!!

**Pull request merged**